pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -126,18 +127,20 @@ func (p *ConfigurationEditFlags) Apply(
 	}
 
 	if cmd.Flags().Changed("env") {
-		envMap, err := util.MapFromArrayAllowingSingles(p.Env, "=")
+		envVars, err := envVarsFromSlice(p.Env, "=")
```

Comment:
Can we move all the env var processing to util ? 

This helps re-using the ordered processing of pairs separated by delimiter, which can be re-used for other configs apart from env vars.